pkg/controller/remotemachineset/azureactuator_test.go

```diff
@@ -0,0 +1,168 @@
+package remotemachineset
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2018-10-01/compute"
+	"github.com/golang/mock/gomock"
+	log "github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/utils/pointer"
+
+	machineapi "github.com/openshift/cluster-api/pkg/apis/machine/v1beta1"
+	azureprovider "sigs.k8s.io/cluster-api-provider-azure/pkg/apis/azureprovider/v1beta1"
+
+	hivev1 "github.com/openshift/hive/pkg/apis/hive/v1"
+	hivev1azure "github.com/openshift/hive/pkg/apis/hive/v1/azure"
+	mockazure "github.com/openshift/hive/pkg/azureclient/mock"
+)
+
+func TestAzureActuator(t *testing.T) {
+	tests := []struct {
+		name                       string
+		mockAzureClient            func(*gomock.Controller, *mockazure.MockClient)
+		clusterDeployment          *hivev1.ClusterDeployment
+		pool                       *hivev1.MachinePool
+		expectedMachineSetReplicas map[string]int64
+		expectedErr                bool
+	}{
+		{
+			name:              "generate single machineset for single zone",
+			clusterDeployment: testAzureClusterDeployment(),
+			pool:              testAzurePool(),
+			mockAzureClient: func(mockCtrl *gomock.Controller, client *mockazure.MockClient) {
+				mockListResourceSKUs(mockCtrl, client, []string{"zone1"})
+			},
+			expectedMachineSetReplicas: map[string]int64{
+				generateAzureMachineSetName("zone1"): 3,
+			},
+		},
+		{
+			name:              "generate machinesets across zones",
+			clusterDeployment: testAzureClusterDeployment(),
+			pool:              testAzurePool(),
+			mockAzureClient: func(mockCtrl *gomock.Controller, client *mockazure.MockClient) {
+				mockListResourceSKUs(mockCtrl, client, []string{"zone1", "zone2", "zone3"})
```

Comment:
Could you add or expand this to have more zones than needed and assert the latter ones are set to 0.